delegator.py

```diff
@@ -151,6 +155,11 @@ def kill(self):
 
     def block(self):
         """Blocks until process is complete."""
+        if self._uses_subprocess:
+            # consume stdout and stderr
+            stdout, stderr = self.subprocess.communicate()
+            self.__out = stdout
+            self.__err = stderr
         self.subprocess.wait()
```

Comment:
Or I guess perhaps it should be inside an `else` if it's used by non-subprocess objects.